File: sorting/QuickSort.py
def partition(l,h):
    pivot = arr[l]
    i=l;j=h
    while(i<j):
        i+=1
        while(arr[i]<=pivot):
            i+=1
        while(arr[j]>pivot):
            j-=1
        if i<j:
            arr[i],arr[j] = arr[j],arr[i]
    arr[l],arr[j]=arr[j],arr[l]
    return j
def QuickSort(l,h):
    while(l<h):
        j = partition(l,h)
        QuickSort(l,j-1)
        QuickSort(j+1,h)
arr = list(map(int,input().split()))
QuickSort(0,len(arr)-1)
for i in arr:
    print(i)

File: sorting/quicksort.cpp
#include<bits/stdc++.h>
using namespace std;
int partition(int arr[], int low, int high){
    int pivot = arr[high];
    int i = low - 1;
    for(int j = low; j < high; j++){
        if(arr[j] <= pivot){
            i++;
            swap(arr[i], arr[j]);
        }
    }
    swap(arr[high], arr[i + 1]);
    return i + 1;
}
void quickSort(int arr[], int low, int high){
    if(low < high){
        int p = partition(arr, low, high);
        quickSort(arr, low, p - 1);
        quickSort(arr, p + 1, high);
    } 
}
int main(){
    int arr[] = {4, 2, 7, 9, 11, 1, 3};
    int n = sizeof(arr)/sizeof(arr[0]);
    quickSort(arr, 0, n - 1);
    for(int i = 0; i < n; i++){
        cout << arr[i] << " ";
    }
    cout << endl;
    return 0;
}